src/screens/Movies.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react';
+import { Text, StyleSheet, TextInput, Button, FlatList, View } from 'react-native';
+import movieService from '../services/MovieService';
+import Movie from '../components/Movie';
+
+export class Movies extends Component {
+  static navigationOptions = {
+    title: 'Movie list'
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      movies: []
+    }
+
+    movieService.getMovies()
```

Comment:
I think that the best place to fetch data is ComponentDidMount. There are two reasons: - "Using DidMount makes it clear that data won’t be loaded until after the initial render. This reminds you to set up initial state properly, so you don’t end up with undefined state that causes errors." and "If you ever need to render your app on the server (SSR/isomorphic/other buzzwords), componentWillMount will actually be called twice – once on the server, and again on the client – which is probably not what you want. Putting the data loading code in componentDidMount will ensure that data is only fetched from the client." https://daveceddia.com/where-fetch-data-componentwillmount-vs-componentdidmount/

Reply:
true, sure - will remember, initializing in the constructor is an old non-react habit